apps/central-scan/frontend/src/components/toggle_test_mode_button.tsx

```diff
@@ -1,98 +1,91 @@
-import { Button, Modal, P } from '@votingworks/ui';
-import React, { useCallback, useRef, useState } from 'react';
-import { useHistory } from 'react-router-dom';
-import { setTestMode } from '../api';
-
-export interface Props {
-  canUnconfigure: boolean;
-  isTestMode: boolean;
-}
+import { Button, Modal, P, SegmentedButton } from '@votingworks/ui';
+import React, { useState } from 'react';
+import { iter } from '@votingworks/basics';
+import { getStatus, getTestMode, setTestMode } from '../api';
 
 /**
  * Presents a button to toggle between test & live modes with a confirmation.
  */
-export function ToggleTestModeButton({
-  canUnconfigure,
-  isTestMode,
-}: Props): JSX.Element {
-  const history = useHistory();
+export function ToggleTestModeButton(): JSX.Element | null {
+  const statusQuery = getStatus.useQuery();
+
+  const testModeQuery = getTestMode.useQuery();
+  const isTestMode = testModeQuery.data ?? false;
+
   const setTestModeMutation = setTestMode.useMutation();
 
-  const [flowState, setFlowState] = useState<
-    'none' | 'confirmation' | 'toggling'
-  >('none');
+  const [flowState, setFlowState] = useState<'none' | 'confirmation'>('none');
   function resetFlowState() {
     setFlowState('none');
   }
 
   function toggleTestMode() {
-    setFlowState('toggling');
     setTestModeMutation.mutate(
       { testMode: !isTestMode },
       {
         onSuccess: () => {
-          history.replace('/');
```

Comment:
I feel like hopping to the main screen after switching mode was a bit weird and unnecessary.